src/contracts/hollowdb.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  get,
+  put,
+  update,
+  remove,
+  evolve,
+  getKeys,
+  getKVMap,
+  updateOwner,
+  updateWhitelist,
+  updateVerificationKey,
+  updateRequirement,
+} from './functions';
+import {InvalidFunctionError} from './errors';
+import type {
+  ContractState,
+  ContractHandle,
+  EvolveInput,
+  GetInput,
+  GetKVMapInput,
+  GetKeysInput,
+  PutInput,
+  RemoveInput,
+  UpdateInput,
+  UpdateOwnerInput,
+  UpdateRequirementInput,
+  UpdateVerificationKeyInput,
+  UpdateWhitelistInput,
+} from './types';
+
+export type HollowState = ContractState<{circuits: ['auth']; whitelists: ['put', 'update']}>;
+export type HollowInput =
+  | GetInput
+  | PutInput
+  | GetKeysInput
+  | GetKVMapInput
+  | UpdateOwnerInput
+  | UpdateWhitelistInput
+  | UpdateVerificationKeyInput
+  | EvolveInput
+  | RemoveInput
+  | UpdateInput
+  | UpdateRequirementInput;
+
+export const handle: ContractHandle<HollowState, HollowInput> = (state, action) => {
```

Comment:
a single `ContractHandle` type takes a state and set of all inputs as type parameter. In doing so, it will ensure the type-safety of everything within.